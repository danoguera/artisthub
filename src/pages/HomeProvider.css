.mainProvContainer h1{
    font-size: 40px;
    text-align: center;
    color: white;
  }

.provServices{
    width: 100%;
   }
   
   .provServices div{
     margin: 16px 16px 0;
     width: calc(100% - 32px);
     height: 300px;
   }
   
   .provService1{
     background-image: url("../assets/images/create.jpg");
     background-repeat: no-repeat;
     background-position: center;
     background-size: cover;
     color: #FFFFFF;
     opacity: 0.8;
   }
   
   .provService1 h3{
     font-size: 50px;
     color: #FFFFFF;
     text-align: center;
     padding-top: 60px;
     line-height: 80px;
     text-shadow: 2px 2px 8px #111;
     font-weight: 900;
     cursor: pointer;
   }
   
   .provService2{
     background-image: url("../assets/images/list.jpg");
     background-repeat: no-repeat;
     background-position: center;
     background-size: cover;
     color: #FFFFFF;
     opacity: 0.9;
   }
   
   .provService2 h3{
     font-size: 50px;
     color: #FFFFFF;
     text-align: center;
     line-height: 230px;
     text-shadow: 2px 2px 8px #111;
     font-weight: 900;
     padding-top: 35px;
   }
   


   
   @media only screen and (min-width: 1000px){

    .provServices div{
        margin: 16px 16px 0;
        width: calc(50% - 32px);
        height: 300px;
        float: left;
      }
    
      .provService2 h3{
        line-height: 200px;
        padding-top: 0px;
      }
    
      .provService1 h3{
        line-height: 200px;
        padding-top: 0px;
      }
   }
   