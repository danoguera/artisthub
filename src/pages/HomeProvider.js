import React from 'react';
import {Link} from 'react-router-dom';
import './HomeProvider.css'


class HomeProvider extends React.Component {
    constructor() {
        super();
        this.state = {

        }
    }

    render() {
        return (

            <React.Fragment>
                <div className="mainProvContainer">
                    <h1>Tell us which service you want to offer:</h1>

                    <div className="provServices">
                        <div className="provService1">
                            <h3><Link to="/posts/create" className="text-link" >Create Service</Link></h3>
                        </div>
                        <div className="provService2">
                            <h3><Link to="/list" className="text-link">List Services</Link></h3>
                        </div>
                    </div>
                </div>
            </React.Fragment>
        )
    }
}

export default HomeProvider;