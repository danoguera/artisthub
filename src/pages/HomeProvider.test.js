import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import { MemoryRouter, Router} from 'react-router-dom';
import { createMemoryHistory } from "history";
import HomeProvider from './HomeProvider';

describe('HomeProvider', () => {

  beforeEach(() => {
    cleanup();
  });

  afterEach(() => {
    cleanup();
  });

  it('should render correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <HomeProvider />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });


   it('should jump to Create Service page page when button is clicked', (done) => {
    const history = createMemoryHistory();
    const { getByText } = render(
        <Router history={history} >
            <HomeProvider/>
        </Router>
    );

      fireEvent.click(getByText(/Create Service/i))
      expect(history.location.pathname).toMatch("/posts/create"); 
      done();

   });

   it('should jump to List posts page when button is clicked', (done) => {
    const history = createMemoryHistory();
    const { getByText } = render(
        <Router history={history} >
            <HomeProvider/>
        </Router>
    );

      fireEvent.click(getByText(/List Services/i))
      expect(history.location.pathname).toMatch("/list"); 
      done();

   });

})