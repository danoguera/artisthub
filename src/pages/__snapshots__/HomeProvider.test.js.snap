// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeProvider should render correctly 1`] = `
<div
  class="mainProvContainer"
>
  <h1>
    Tell us which service you want to offer:
  </h1>
  <div
    class="provServices"
  >
    <div
      class="provService1"
    >
      <h3>
        <a
          class="text-link"
          href="/posts/create"
        >
          Create Service
        </a>
      </h3>
    </div>
    <div
      class="provService2"
    >
      <h3>
        <a
          class="text-link"
          href="/list"
        >
          List Services
        </a>
      </h3>
    </div>
  </div>
</div>
`;
